.ci_support/migrations/cuda120.yaml

```diff
@@ -0,0 +1,80 @@
+    cuda_compiler_version:
+      - None
+      - 10.2                       # [(linux64 or win) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+      - 11.0                       # [(linux64 or win) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+      - 11.1                       # [(linux64 or win) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+      - 11.2                       # [(linux64 or win) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+      - 12.0                       # [(linux64 or win) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  commit_message: |
+    Rebuild for CUDA 12
+    
+    The transition to CUDA 12 SDK includes new packages for all CUDA libraries and
+    build tools. Notably, the cudatoolkit package no longer exists, and packages
+    should depend directly on the specific CUDA libraries (libblas, libcusolver,
+    etc) as needed. For an in-depth overview of the changes and to report problems
+    [see this issue]( https://github.com/conda-forge/conda-forge.github.io/issues/1963 ).
+    Please feel free to raise any issues encountered there. Thank you! :pray:
+
+cuda_compiler:                 # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  - cuda-nvcc                  # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+
+cuda_compiler_version:         # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  - 12.0                       # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+
+c_compiler_version:            # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  - 12                         # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+
+cxx_compiler_version:          # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  - 12                         # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+
+fortran_compiler_version:      # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  - 12                         # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+
+cudnn:                         # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  - 8                          # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+
+cdt_name:                      # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  - cos7                       # [linux64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+
+docker_image:                                      # [os.environ.get("BUILD_PLATFORM", "").startswith("linux-") and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  - quay.io/condaforge/linux-anvil-cos7-x86_64     # [linux64 and os.environ.get("BUILD_PLATFORM") == "linux-64" and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
```

Comment:
The problem is here. This is not specifying an image for cross-compilation

Reply:
Though this is just an `x86_64` migrator. Cross-compilation images only become relevant when [the CUDA 12 arch migrator is enabled]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/migrations/cuda120_ppc64le_aarch64.yaml#L9-L10 )

Reply:
The lack of proper image is the direct cause of the mis-render. As shown by https://github.com/conda-forge/arrow-cpp-feedstock/pull/1085/commits/a8013b27fe28e4f198787ead51ae44017142d25c & rerendering.

Whether that's a bug in the migrator infra or an oversight in the migrator, it seems like the pragmatic way to solve this (at least in the context of this feedstock)

Reply:
Should we make that change in the migrator in `conda-forge-pinning` then?

Reply:
IMO yes

Reply:
Though I admit it's interesting that 10.x and 11.x didn't need this, even though they're both also cross-compiling CUDA.

Reply:
Then again, the recipe output structure was majorly overhauled for 10.x, so I can _kinda_ see how that might influence the rerender, but it's still weird.